### PR TITLE
Cleanup gradle files

### PR DESCRIPTION
**Pull request by DRSchlaubi**
The Gradle configuration showed a lot of duplicated code, this PR aims to fix that﻿

### CODE REVIEW

**Comment by gdude2002** [comment]
This looks considerably better, but can you quickly ELI5 what you've done here? It looks like you've created three gradle plugins that handle the common tasks, right?

**Comment by DRSchlaubi** [comment]
I have no idea what ELI5 means so I assume you want a quick TL;DR

If you craete a gradle.kts in buildSrc Gradle will generate a plugin implimentation for you, which applies that buildscript as a plugin.

So here q quick explanation about what each one does and how you can configure it

- `kordex-module`:
  desc: top level plugin
  configures: kotlin (jvmtarget and java plugin target), detekt config, sources and javadoc jars, explicitApi mode
  config: if you don't target java 8 or want to disable explicitApi, you have the Kordex extension

```kotlin
kordex {
    jvmTarget.set("9")
    javaVersion.set(JavaVersion.VERSION_1_9)
    useExplicitApiMode.set(false)
}
```

- `tested-module`
  configures: test task (**THIS DOESN'T ADD JUNIT DEPS**, because you can't use dependency catalogs in buildSrc)

- `ksp-module`
 configures: ksp source sets (this adds the ksp generated output to the main source set, ~~it doesn't add the plugin for ksp~~ though)
- `dokka-module`
  configures: dokkaHtml task
  configuration: If you want to set a moduleName, or include md files

```kotlin
dokkaModule {
    moduleName.set("Kord Extensions: Annotation Processor")
    includes.add("packages.md")
}
```

- `published-module.`
  desc: configures publishing
  configures: signing and maven publish tasks

**Comment by gdude2002** [comment]
Hmm, okay, that makes a lot of sense. Any reason the KSP plugin can't be applied by `ksp-module`? There'll be no KSP output without the plugin regardless.

**Comment by DRSchlaubi** [comment]
> Hmm, okay, that makes a lot of sense. Any reason the KSP plugin can't be applied by `ksp-module`? There'll be no KSP output without the plugin regardless.

no reason it can't, it just wasn't applied to all modules having the sourceSets

**Comment by gdude2002** [comment]
You can chalk that up to poor copypasta, lmao

**Comment by DRSchlaubi** [comment]
> You can chalk that up to poor copypasta, lmao

Okay then'll remove the dokka-module from all modules not having the plugin applied and add the plugin directly in the module

**Comment by gdude2002** [comment]
Good idea, appreciate that

**Comment by DRSchlaubi** [comment]
Okay one last problem seems to be that intellij doesn't pick up the kotlin config correctly I will try to fix that real quick

**Comment by gdude2002** [comment]
We good?

**Comment by DRSchlaubi** [comment]
We should be

**Comment by gdude2002** [comment]
🚀 

Thanks for looking into this!
